packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -793,6 +793,210 @@ describe("Controller helper integration test", function () {
+      const [amount0, amount1] = await (positionManager as INonfungiblePositionManager).connect(depositor).callStatic.decreaseLiquidity({
+        tokenId: tokenId,
+        liquidity: position.liquidity,
+        amount0Min: 0,
+        amount1Min: 0,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+      })
+      const wPowerPerpAmountInLP = (isWethToken0) ? amount1 : amount0;
+      const wethAmountInLP = (isWethToken0) ? amount0 : amount1;
+
+      // console.log('position.tickLower', position.tickLower.toString())
+      // console.log('position.tickUpper', position.tickUpper.toString())
+      // console.log('ownerOfUniNFT', ownerOfUniNFT)
+      // console.log('depositor.address', depositor.address)
+      // console.log('wPowerPerpAmountInLP',wPowerPerpAmountInLP.toString())
+      // console.log('wethAmountInLP',wethAmountInLP.toString())
+      // console.log('vaultBefore.shortAmount',vaultBefore.shortAmount.toString())
+      // console.log('vaultBefore.collateralAmount',vaultBefore.collateralAmount.toString())
+      // console.log('vaultAfter.shortAmount',vaultAfter.shortAmount.toString())
+      // console.log('vaultAfter.collateralAmount',vaultAfter.collateralAmount.toString())
+      // console.log('depositorEthBalanceBefore',depositorEthBalanceBefore.toString())
+      // console.log('depositorEthBalanceAfter',depositorEthBalanceAfter.toString())
+      // console.log('collateralAmount',collateralAmount.toString())
+      // console.log('gasSpent',gasSpent.toString())
+      // console.log('depositorEthBalanceAfter',depositorEthBalanceAfter.toString())
+      // console.log('collateralToLp',collateralToLp.toString())
+      // console.log('mintWSqueethAmount',mintWSqueethAmount.toString())
+      // console.log('tokenIndexBefore',tokenIndexBefore.toString())
+      // console.log('tokenIndexAfter',tokenIndexAfter.toString())
+
+      expect(position.tickLower === newTickLower).to.be.true
+      expect(position.tickUpper === newTickUpper).to.be.true
+      expect(ownerOfUniNFT === depositor.address).to.be.true
+      expect(tokenIndexAfter.sub(tokenIndexBefore).eq(BigNumber.from(1))).to.be.true
+      expect(vaultBefore.shortAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultBefore.collateralAmount.eq(BigNumber.from(0))).to.be.true
+      expect(vaultAfter.collateralAmount.eq(collateralAmount)).to.be.true
+      // One side LP - weth should be zero
+      expect(wPowerPerpAmountInLP.sub(mintWSqueethAmount).abs().lte(10)).to.be.true
+      expect(wethAmountInLP.eq(BigNumber.from(0))).to.be.true
+      //uniswap rounding of LP often gives some wei less than expected (uniswap takes the wei, but only gives credit for less wei in the LP share)
+      expect((depositorEthBalanceBefore.sub(depositorEthBalanceAfter).sub(collateralAmount).sub(wethAmountInLP).sub(gasSpent)).abs().lte(1)).to.be.true
+      expect((vaultAfter.shortAmount.sub(wPowerPerpAmountInLP)).abs().lte(1)).to.be.true
```

Comment:
think should check that wPowerPerpAmountInLp is what we expect it to be not just that its equal to minted amount in vault

Reply:
ah yeah thats good